### ISSUE TITLE
Source build produces HarmonicBondForce error for AMOEBA

### ISSUE THREAD

**Issue by Gareth-elliott** [openmm/openmm#3304]
Hi @peastman,

I built OpenMM from source at the master branch yesterday (wanting to test simulations with the updated AMOEBA features). It built fine, and I ran the tests, with one failing (TestCudaCompiler, see output below), but since all the other tests worked, I wasn't worried about that.  I then tried testing the code, and got the following error:

HarmonicBondForce: force constant cannot be negative

The error happens when an old version of the amoeba force field (amoeba2013.xml), but also happens in the case where I'm trying to use amoeba2018.xml, and I pass "rigidWater=False" to forcefield.createSystem(...). Removing this appears to make it run fine (I had it in my script from when I initially set it up because I was wasn't sure exactly how it was setting it and wanted to avoid ambiguity)

I'll attach a .pdb file and a .py file that tests the cases (and outputs the stack trace) so you can have a look.

[system.pdb.txt](https://github.com/openmm/openmm/files/7431354/system.pdb.txt)
[omm_test.py.txt](https://github.com/openmm/openmm/files/7431356/omm_test.py.txt)
Thanks,

Gareth

**ctest output**
ctest --rerun-failed --output-on-failure
Test project /home/c3162723/software/src/my_openmm_fork/openmm/build_27Oct2021
    Start 407: TestCudaCompiler
1/1 Test #407: TestCudaCompiler .................***Failed    1.12 sec
sh: this does not exist/openmmTempKernel0x71b4c0_37453.log: No such file or directory
exception: Error launching CUDA compiler: 256

**Comment by peastman** [comment]
Thanks!  This was caused by #3286, where I added lots of checks for obviously wrong parameter values.  Because why could anyone possible want a negative spring constant for a harmonic bond, right?  Well, it turns out AMOEBA does!  It uses it for the Urey-Bradley term on water molecules.  A small length term with negative spring constant acts as a correction to a larger angle term with positive spring constant.

I guess I'll remove that check.

Your test also revealed another problem with AMOEBA18: it's supposed to set `rigidWater=False` by default.  I'll create a PR fixing both these problems.
